pkgs/applications/science/molecular-dynamics/lammps/default.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv, writeText, fetchurl,
+  libpng, fftw,
+  mpiSupport ? false, mpi ? null
+}:
+
+assert mpiSupport -> mpi != null;
+
+stdenv.mkDerivation rec {
+  version = "16Feb16";
+  name = "lammps-${version}";
+
+  src = fetchurl {
+    url = "http://downloads.sourceforge.net/project/lammps/${ name }.tar.gz";
```

Comment:
`fetchurl` supports mirror URLs for sourceforge: `mirror://sourceforge/lammps/${name}.tar.gz` should work. (Also note your original has a typo).